rpc/send_test.go

```diff
@@ -241,28 +243,27 @@ func TestClientNotReady(t *testing.T) {
 	// Send the RPC again with no timeout.
 	opts.SendNextTimeout = 0
 	opts.Timeout = 0
-	c := make(chan struct{})
+	c := make(chan error)
 	go func() {
 		if _, err := sendPing(opts, []net.Addr{ln.Addr()}, nodeContext); err == nil {
-			t.Fatalf("expected error when client is closed")
+			c <- fmt.Errorf("expected error when client is closed")
```

Comment:
It's a data race to call `t.Fatalf` from a goroutine other than the one that started the test.